src/l500/l500-color.cpp

```diff
@@ -493,10 +493,44 @@ namespace librealsense
         return tags;
     }
 
-    std::vector<uint8_t> l500_color::get_raw_intrinsics_table() const
+    ivcam2::intrinsic_rgb l500_color::get_intrinsics_table() const
     {
-        AC_LOG( DEBUG, "RGB_INTRINSIC_GET" );
-        return _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+        // Get RAW data from firmware
+        std::vector< uint8_t > response_vec = _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+
+        if (response_vec.size() == 0)
+            throw invalid_value_exception("Calibration data invalid,buffer size is zero");
+
+        auto resolutions_rgb_table_ptr = reinterpret_cast<const ivcam2::intrinsic_rgb *>(response_vec.data());
```

Comment:
The raw table must be validated against the recognized format template:
`auto intrinsic = check_calib<intrinsic_rgb>( *_owner->_color_intrinsics_table_raw );`

Reply:
I don't know what check_calib is meant to do but from what I see it only checks for size. I don't think it's readable or adds anything. We have a function that reads a table, it can check the calibration here, no? I can see separating the check into another function, but not with a template, and the cast would still be made here.

Reply:
The function tries to parse the struct according to the predefined header and also verify the buffer's binary integrity (CRC)
It is crucial to ensure data correctness. Why this was capability is removed ? 

The template is required as calib table for all sensor are using the header structure.

Reply:
> The function tries to parse the struct according to the predefined header and also verify the buffer's binary integrity (CRC). It is crucial to ensure data correctness. Why this capability was removed ?
> 
> The template is required as calib table for all sensor are using the header structure.

This is true for DS5 device.
On L515 device we have no CRC value inside the data and no header also.
+ according to FW API guidelines, The size of the output will be >= to the expected one.

The validation can only count on the implementation I did. calculated expected size <= output_size